README.md

```diff
@@ -13,40 +13,46 @@ It also contains resources useful to OpenCL developers:
 
-To clone a new repository with all sub-modules included, use the `--recursive` option.
-Note that this option clones all sub-modules and their dependencies, which are not required for the OpenCL SDK:
+- This repository uses sub-modules for the OpenCL Headers, OpenCL C++ bindings, and OpenCL ICD Loader and some of their transitive dependencies.
 
-```sh
-$ git clone --recursive https://github.com/KhronosGroup/OpenCL-SDK.git
-```
+  - To clone a new repository with all sub-modules included, use the `--recursive` option. Note that this option clones all sub-modules and their dependencies, which are not strictly required for the OpenCL SDK:
 
-Alternatively, to clone only the sub-modules for the OpenCL SDK, first clone this repository without sub-modules included:
+        git clone --recursive https://github.com/KhronosGroup/OpenCL-SDK.git
 
-```sh
-$ git clone https://github.com/KhronosGroup/OpenCL-SDK.git
-```
+  - Alternatively, to clone only the sub-modules for the OpenCL SDK, first clone this repository without sub-modules included then setup submodules non-recursively:
 
-Then setup sub-modules:
+        git clone https://github.com/KhronosGroup/OpenCL-SDK.git
+        git submodule init
+        git submodule update
 
-```sh
-$ git submodule init
-$ git submodule update
-```
+- The SDK uses CMake for its build system.
+If CMake is not provided by your build system or OS package manager, please consult the [CMake website](https://cmake.org).
 
-## Building the Samples
+- The SDK samples depend on
 
-This repository uses CMake as its build system.
-The suggested build directory is `build`.
+  - [Templatized C++ Command Line Parser Library](http://tclap.sourceforge.net/) (aka. TCLAP)
+  - [Simple and Fast Multimedia Library](https://www.sfml-dev.org/) (aka. SFML)
+  - [OpenGL Mathematics](https://glm.g-truc.net/0.9.9/index.html) (aka. GLM)
 
-To generate build files, use for example:
+### Example Build
 
-```sh
-$ mkdir build && cd build
-$ cmake ..
-```
+> The example build guide uses [Vcpkg](https://vcpkg.io/en/index.html) to fetch all dependencies. Note that Vcpkg is _not_ a requirement and is only used for convenience. One may provide dependencies through any other CMake mechanism. For details on how to install Vcpkg, refer to it's [Getting Started Guide](https://vcpkg.io/en/getting-started.html)
 
-Then build with the generated build files.
+1. Clone this repo with the rest of the OpenCL SDK components:
+
+       git clone https://github.com/KhronosGroup/OpenCL-SDK.git
+       git submodule init
+       git submodule update
+
+1. Install dependencies
+
+       vcpkg install sfml tclap glm
+
+1. Build and install SDK with samples and no downstream unit tests
+
+       cmake -D BUILD_TESTING=OFF -D BUILD_DOCS=OFF -D BUILD_EXAMPLES=OFF -D BUILD_TESTS=OFF -D OPENCL_SDK_BUILD_SAMPLES=ON -D OPENCL_SDK_TEST_SAMPLES=OFF -D CMAKE_TOOLCHAIN_FILE=/vcpkg/install/root/scripts/buildsystems/vcpkg.cmake -B ./OpenCL-SDK/build -S ./OpenCL-SDK
```

Comment:
Suggest breaking up this long line to ease readability:

```suggestion
       cmake -D BUILD_TESTING=OFF \
               -D BUILD_DOCS=OFF \
               -D BUILD_EXAMPLES=OFF \
               -D BUILD_TESTS=OFF \
               -D OPENCL_SDK_BUILD_SAMPLES=ON \
               -D OPENCL_SDK_TEST_SAMPLES=OFF \
               -D CMAKE_TOOLCHAIN_FILE=/vcpkg/install/root/scripts/buildsystems/vcpkg.cmake \
               -B ./OpenCL-SDK/build -S ./OpenCL-SDK
```

It might also be a good idea to use a generic path like `/path/to/your/vcpkg.cmake` since this will change depending on how vcpkg is installed.

Reply:
Ah, thanks. I'm still at war with the single-line, multi-line YAML syntaxes of GitLab/GitHub. Sprinkle some Dockerfile syntax on top and confusion is maxed out. I just accepted my fate, but this is nicer.